tests/discrete_domain.cpp

```diff
@@ -129,6 +133,16 @@ TEST(ProductMDomainTest, Diff)
     EXPECT_EQ(subdomain, dom_x);
 }
 
+TEST(ProductMDomainTest, Replace)
+{
+    DDomXY const dom_x_y = DDomXY();
+    DDomZ const dom_z = DDomZ();
+    DDomXZ const dom_x_z = DDomXZ();
```

Comment:
Can you give them different sizes to be sure this information gets also correctly replaced in `subdomain`.